linkerd/app/gateway/src/lib.rs

```diff
@@ -116,11 +116,15 @@ where
         .push_tcp_logical(resolve.clone())
         .into_stack()
         .push_request_filter(|(p, _): (Option<profiles::Receiver>, _)| match p {
-            Some(rx) if rx.borrow().addr.is_some() => Ok(outbound::tcp::Logical {
-                profile: Some(rx),
-                orig_dst: OrigDstAddr(std::net::SocketAddr::from(([0, 0, 0, 0], 0))),
-                protocol: (),
-            }),
+            Some(rx) if rx.borrow().addr.is_some() => {
+                let logical_addr = rx.borrow().addr.clone();
+                return Ok(outbound::tcp::Logical {
```

Comment:
looks like Clippy doesn't like the unnecessary `return` here:
```suggestion
                Ok(outbound::tcp::Logical {
```